src/kinetics/GasKinetics.cpp

```diff
@@ -54,6 +54,18 @@ void GasKinetics::update_rates_T()
             m_ROP_ok = false;
         }
     }
+
+    double Te = thermo().electronTemperature();
+    double logTe = log(Te);
+
+    if (T != m_temp || Te != m_temp_e) {
+        if (m_electron_temperature_rates.nReactions()) {
+            m_electron_temperature_rates.update(T, Te, logTe, m_rfn.data());
+            m_ROP_ok = false;
+        }
+    }
```

Comment:
I left m_electron_temperature_rates because you can manually set electron temperature without using PlasmaPhase. Maybe this will be useful for simulations with external calculated electron temperature.

Reply:
I can see the logic in that, but I'd like to avoid increasing the complexity of the widely-used `GasKinetics` class for what I think is a relatively special case. As an alternative, what about moving this to `PlasmaKinetics` class as I suggested, but making it so that `PlasmaKinetics` can be used with IdealGasPhase as well?

Since the `dynamic_cast` in `PlasmaKinetics::init` will return 0 if `thermo` isn't a `PlasmaPhase`, all you have to do to make this work is check whether `m_plasma` is 0 in `PlasmaKinetics::addPlasmaReaction`, and raise an exception (e.g. "Plasma reaction type requires the a thermo model derived from PlasmaPhase") if it is. Then, you could still add "electron temperature" reactions with this class, and the existing implementation of `PlasmaKinetics::update_rates_T` will still work, since `m_plasmaProcessIndx` will be empty if there are no `PlasmaReaction`s.